okhttp/src/main/kotlin/okhttp3/internal/concurrent/TaskRunner.kt

```diff
@@ -243,14 +243,16 @@ class TaskRunner(
   }
 
   fun cancelAll() {
-    for (i in busyQueues.size - 1 downTo 0) {
-      busyQueues[i].cancelAllAndDecide()
-    }
-    for (i in readyQueues.size - 1 downTo 0) {
-      val queue = readyQueues[i]
-      queue.cancelAllAndDecide()
-      if (queue.futureTasks.isEmpty()) {
-        readyQueues.removeAt(i)
+    synchronized(this) {
```

Comment:
Please replace `synchronized`  with this:
```
    this.assertThreadHoldsLock()
```
Our only production caller holds a lock when it calls this; I don’t want re-entrant synchronized blocks if I can avoid it.

Possibly also requires us to change `OkHttpClientTestRule` to acquire the lock before making this call.

In either case, DO NOT call APIs in `okhttp3.internal`; they’re undocumented, subject to change, and not for end users’ use.

Reply:
Ok.